core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -135,10 +141,17 @@
 	 * @param resolver
 	 */
 	protected SailSourceConnection(AbstractSail sail, SailStore store, FederatedServiceResolver resolver) {
+		this(sail, store, new SimpleEvaluationStrategyFactory(resolver), resolver);
```

Comment:
That will mean that only Sail implementations that rely on SailSourceConnection can actually make use of this - which is not the design intent. The overall design of the various APIs has always been that functionality such as SPARQL query evaluation is _provided_ to the SAIL implementation, and it can choose to reuse or override. These evaluation "modes" are core functionality that the query engine should provide for _any_ SAIL implementation out of the box.